test/functional/feature_assumeutxo.py

```diff
@@ -62,6 +66,25 @@ def setup_network(self):
         self.add_nodes(3)
         self.start_nodes(extra_args=self.extra_args)
 
+    def test_invalid_snapshot_scenarios(self, valid_snapshot_path):
+        self.log.info("Test different scenarios of loading invalid snapshot files")
+        self.log.info("  - snapshot file refering to a block that is not in the assumeutxo parameters")
+        with open(valid_snapshot_path, 'rb') as f:
+            valid_snapshot_contents = f.read()
+
+        # we can only test this with a block that is already known, as otherwise the `loadtxoutset` RPC
+        # would time out (waiting to see the hash in the headers chain), rather than error immediately
+        bad_snapshot_height = SNAPSHOT_BASE_HEIGHT - 1
+        bad_snapshot_path = valid_snapshot_path + '.mod'
+        with open(bad_snapshot_path, 'wb') as f:
+            bad_snapshot_block_hash = self.nodes[0].getblockhash(bad_snapshot_height)
+            # block hash of the snapshot base is stored right at the start (first 32 bytes)
+            f.write(bytes.fromhex(bad_snapshot_block_hash)[::-1] + valid_snapshot_contents[32:])
+
+        expected_log = f"assumeutxo height in snapshot metadata not recognized ({bad_snapshot_height}) - refusing to load snapshot"
+        with self.nodes[1].assert_debug_log(expected_log):
```

Comment:
Good catch. Unfortunately, I just noticed that the exact same mistake also happened in #28634 :scream:  . Given that this is really hard to catch (in total five out of six reviewers didn't see it), it may make sense to assert the type in `assert_debug_log` to avoid bugs like that in the future.